pkg/core/chain/chain.go

```diff
@@ -482,6 +482,23 @@ func (c *Chain) postAcceptBlock(blk block.Block, l *logrus.Entry) {
 		go c.storeStakesInStormDB(blk.Header.Height)
 	}
 
+	// 4. Persist state
+	pe := config.Get().State.PersistEvery
+
+	if pe > 0 {
+		if blk.Header.Height%pe == 0 {
+			l.Debug("chain: persist state", blk.Header.Height)
+
+			if err := c.proxy.Executor().Persist(c.ctx, c.tip.Header.StateHash); err != nil {
+				// TODO: trigger resync procedure
+				// https://github.com/dusk-network/dusk-blockchain/issues/1286
+				l.WithError(err).Error("chain: persist state failed")
+			}
+		}
+	} else {
+		l.Warn("State won't persist! Set `state.persistEvery` to a positive value")
```

Comment:
That's kind of the idea, one should never have this value at `0`. But, I get that for some random test it could be useful. I see a couple things we could do:
1. Add a "counter" member variable to `Chain` that helps us print "every _n_".
2. Have a RNG and have it print with a 20% probability...

First case I dislike, putting a member variable just for a `warn` print is an ugly idea. Second case I like better, but it introduces overhead. What do you think?